frontend/libs/studio-components/src/components/StudioPageHeader/StudioPageHeaderProfileMenu/StudioPageHeaderProfileMenu.tsx

```diff
@@ -0,0 +1,169 @@
+    </DropdownMenu>
+  );
+};
+
+type StudioPageHeaderMenuContentProps = {
+  profileMenuGroups: StudioProfileMenuGroup[];
+  onClickItem: () => void;
+};
+const StudioPageHeaderMenuContent = ({
+  profileMenuGroups,
+  onClickItem,
+}: StudioPageHeaderMenuContentProps): ReactElement => {
+  return (
+    <DropdownMenu.Content>
+      {profileMenuGroups.map((group: StudioProfileMenuGroup, index: number) => (
+        <StudioPageHeaderMenuContentGroup key={index} group={group} onClickItem={onClickItem} />
+      ))}
+    </DropdownMenu.Content>
+  );
+};
+
+type StudioPageHeaderMenuContentGroupProps = {
+  group: StudioProfileMenuGroup;
+  onClickItem: () => void;
+};
+const StudioPageHeaderMenuContentGroup = ({
+  group,
+  onClickItem,
+}: StudioPageHeaderMenuContentGroupProps): ReactElement => {
+  return (
+    <DropdownMenu.Group className={classes.dropDownMenuGroup}>
+      {group.items.map((item: StudioProfileMenuItem) => (
+        <StudioProfileMenuGroupItem key={item.itemName} item={item} onClickItem={onClickItem} />
+      ))}
+    </DropdownMenu.Group>
+  );
+};
+
+type StudioProfileMenuGroupItemProps = {
+  item: StudioProfileMenuItem;
+  onClickItem: () => void;
+};
+const StudioProfileMenuGroupItem = ({
+  item,
+  onClickItem,
+}: StudioProfileMenuGroupItemProps): ReactElement => {
+  const handleClickMenuItemButton = (menuItem: StudioProfileMenuItem) => {
+    menuItem.action.type === 'button' && menuItem.action.onClick();
+    onClickItem();
+  };
+
+  if (item.action.type === 'button') {
+    return (
+      <StudioProfileMenuButton
+        key={item.itemName}
+        itemName={item.itemName}
+        isActive={item.isActive}
+        onClick={() => handleClickMenuItemButton(item)}
+      />
+    );
+  }
+  return (
+    <StudioProfileMenuLink
+      key={item.itemName}
+      itemName={item.itemName}
+      href={item.action.href}
+      openInNewTab={item.action.openInNewTab}
+    />
+  );
+};
+
+type StudioProfileMenuButtonProps = {
+  itemName: string;
+  isActive?: boolean;
+} & DropdownMenuItemProps;
+
+const StudioProfileMenuButton = ({
+  itemName,
+  isActive,
+  ...rest
+}: StudioProfileMenuButtonProps): ReactElement => {
+  return (
+    <DropdownMenu.Item
+      key={itemName}
+      className={classes.menuItemButton}
+      aria-checked={isActive}
+      role='menuitemradio'
+      {...rest}
+    >
+      {itemName}
+    </DropdownMenu.Item>
+  );
+};
+
+type StudioProfileMenuLinkProps = {
+  itemName: string;
+  href: string;
+  openInNewTab?: boolean;
+};
+
+const StudioProfileMenuLink = ({
+  itemName,
+  href,
+  openInNewTab,
+}: StudioProfileMenuLinkProps): ReactElement => {
+  return (
+    <DropdownMenu.Item key={itemName} asChild>
+      <a
+        href={href}
+        target={openInNewTab ? '_blank' : undefined}
+        rel={openInNewTab ? 'noopener noreferrer' : undefined}
+      >
+        {itemName}
+      </a>
+    </DropdownMenu.Item>
+  );
+};
```

Comment:
`handleClickMenuItemButton` brukes bare på knapper, så den kan vi definere inne i den komponenten. Vi trenger heller ikke å sende inn `item` siden den allerede er tilgjengelig der. `key` er heller ikke nødvendig på rotkomponentene. Her er et komplett forslag til hvordan vi kan endre koden for å ta høyde for det. Denne bygger også på typeendringen som jeg foreslo i den forrige kommentaren.
```suggestion
type StudioProfileMenuGroupItemProps = {
  item: StudioProfileMenuItem;
  onClickItem: () => void;
};

const StudioProfileMenuGroupItem = ({
  item,
  onClickItem,
}: StudioProfileMenuGroupItemProps): ReactElement =>
  item.action.type === 'button' ? (
    <StudioProfileMenuButton
      item={item as StudioProfileMenuItem<'button'>}
      onClickButton={onClickItem}
    />
  ) : (
    <StudioProfileMenuLink item={item as StudioProfileMenuItem<'link'>} />
  );

type StudioProfileMenuButtonProps = {
  item: StudioProfileMenuItem<'button'>;
  onClickButton: () => void;
} & DropdownMenuItemProps;

const StudioProfileMenuButton = ({
  item,
  onClickButton,
  ...rest
}: StudioProfileMenuButtonProps): ReactElement => {
  const handleClick = () => {
    item.action.onClick();
    onClickButton();
  };

  return (
    <DropdownMenu.Item
      className={classes.menuItemButton}
      aria-checked={item.isActive}
      role='menuitemradio'
      onClick={handleClick}
      {...rest}
    >
      {item.itemName}
    </DropdownMenu.Item>
  );
};

type StudioProfileMenuLinkProps = {
  item: StudioProfileMenuItem<'link'>;
};

const StudioProfileMenuLink = ({ item }: StudioProfileMenuLinkProps): ReactElement => {
  const { href, openInNewTab } = item.action;
  return (
    <DropdownMenu.Item asChild>
      <a
        href={href}
        target={openInNewTab ? '_blank' : undefined}
        rel={openInNewTab ? 'noopener noreferrer' : undefined}
      >
        {item.itemName}
      </a>
    </DropdownMenu.Item>
  );
};
```